src/main/webapp/app/iris/manage/settings/shared/iris-enabled/iris-enabled.component.spec.ts

```diff
@@ -83,14 +86,120 @@ describe('IrisEnabledComponent', () => {
+        componentRef.setInput('irisSubSettingsType', IrisSubSettingsType.ALL);
+        comp.irisSettings = {
+            ...irisSettings,
+            irisChatSettings: { type: IrisSubSettingsType.CHAT, enabled: true },
+            irisTextExerciseChatSettings: { type: IrisSubSettingsType.TEXT_EXERCISE_CHAT, enabled: true },
+            irisCourseChatSettings: { type: IrisSubSettingsType.COURSE_CHAT, enabled: true },
+            irisCompetencyGenerationSettings: { type: IrisSubSettingsType.COMPETENCY_GENERATION, enabled: true },
+            irisLectureChatSettings: { type: IrisSubSettingsType.LECTURE, enabled: true },
+            irisFaqIngestionSettings: { type: IrisSubSettingsType.FAQ_INGESTION, enabled: true, autoIngestOnFaqCreation: false },
+            irisLectureIngestionSettings: { type: IrisSubSettingsType.LECTURE_INGESTION, enabled: true, autoIngestOnLectureAttachmentUpload: false },
+        };
+        fixture.detectChanges();
+
+        comp.setEnabled(false);
+
+        expect(setSettingsSpy).toHaveBeenCalledOnce();
+        expect(comp.irisSettings.irisChatSettings?.enabled).toBeFalse();
+        expect(comp.irisSettings.irisTextExerciseChatSettings?.enabled).toBeFalse();
+        expect(comp.irisSettings.irisCourseChatSettings?.enabled).toBeFalse();
+        expect(comp.irisSettings.irisCompetencyGenerationSettings?.enabled).toBeFalse();
+        expect(comp.irisSettings.irisLectureChatSettings?.enabled).toBeFalse();
+        expect(comp.irisSettings.irisFaqIngestionSettings?.enabled).toBeFalse();
+        expect(comp.irisSettings.irisLectureIngestionSettings?.enabled).toBeFalse();
+        expect(comp.someButNotAllSettingsEnabled).toBeFalse();
+    });
+
+    it('should create new subsettings if they do not exist when IrisSubSettingsType is ALL and setEnabled is called', async () => {
+        const setSettingsSpy = jest.spyOn(irisSettingsService, 'setCourseSettings').mockReturnValue(of(new HttpResponse({ body: null as any as IrisSettings })));
+        jest.spyOn(irisSettingsService, 'getUncombinedCourseSettings').mockReturnValue(of({} as IrisCourseSettings));
+        componentRef.setInput('course', course);
+        componentRef.setInput('irisSubSettingsType', IrisSubSettingsType.ALL);
+        comp.irisSettings = { type: IrisSettingsType.COURSE };
+        fixture.detectChanges();
+
+        comp.setEnabled(true);
+
+        expect(setSettingsSpy).toHaveBeenCalledOnce();
+        expect(comp.irisSettings!.irisChatSettings).toBeDefined();
+        expect(comp.irisSettings!.irisTextExerciseChatSettings).toBeDefined();
+        expect(comp.irisSettings!.irisCourseChatSettings).toBeDefined();
+        expect(comp.irisSettings!.irisCompetencyGenerationSettings).toBeDefined();
+        expect(comp.irisSettings!.irisLectureChatSettings).toBeDefined();
+        expect(comp.irisSettings!.irisFaqIngestionSettings).toBeDefined();
+        expect(comp.irisSettings!.irisLectureIngestionSettings).toBeDefined();
```

Comment:
Are these necessary? The "enabled" checks below should fail, too in these cases.